According to the manual page, without --webpage or --continous no pages are produced, which is not what we want.
diff -Naur chm2pdf-0.9/chm2pdf chm2pdf-0.9-p/chm2pdf
--- chm2pdf-0.9/chm2pdf	2007-11-18 15:24:11.000000000 +0100
+++ chm2pdf-0.9-p/chm2pdf	2008-05-09 01:39:25.000000000 +0200
@@ -415,8 +415,8 @@
             elif key=='version': htmldoc_opts += ' ' + value
             elif key=='webpage': htmldoc_opts += ' ' + value
            
-    print 'htmldoc' + htmldoc_opts + ' ' + htmlout_filename_list + " -f "+ outputfilename + " > /dev/null"
-    os.system ('htmldoc' + htmldoc_opts + ' ' + htmlout_filename_list + " -f "+ outputfilename + " > /dev/null")
+    print 'htmldoc' + htmldoc_opts + ' ' + htmlout_filename_list + " -f "+ outputfilename + " --webpage > /dev/null"
+    os.system ('htmldoc' + htmldoc_opts + ' ' + htmlout_filename_list + " -f "+ outputfilename + " --webpage > /dev/null")
 
     print 'Written file ' + outputfilename
     print 'Done.'
